.github/workflows/deploy.yml

```diff
@@ -0,0 +1,75 @@
+name: Deploy
+
+on:
+  release:
+    types: [created]
+
+defaults:
+  run:
+    shell: pwsh
+
+env:
+  baseVersion: 1.0.0
+
+jobs:
+  build-and-test:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - name: Setup .NET
+      uses: actions/setup-dotnet@v2
+      with:
+        dotnet-version: | 
+            2.1.x
+            3.1.x
+            6.x
+    - name: Restore dependencies
+      run: dotnet restore
+    - name: Set Version
+      run: |
+        echo "${{ github.ref }}"
+        if ("${{ github.ref }}".startsWith("refs/tags/v")) {
+          $tagVersion = "${{ github.ref }}".substring(11)
+          echo "buildVersion=$tagVersion.${{ github.run_number }}" | Out-File -FilePath $env:GITHUB_ENV -Encoding utf8 -Append
+          echo "nugetVersion=$tagVersion" | Out-File -FilePath $env:GITHUB_ENV -Encoding utf8 -Append
+          echo "preRelease=false" | Out-File -FilePath $env:GITHUB_ENV -Encoding utf8 -Append
+        } else {
+          echo "buildVersion=${{ env.baseVersion }}.${{ github.run_number }}" | Out-File -FilePath $env:GITHUB_ENV -Encoding utf8 -Append
+          echo "nugetVersion=${{ env.baseVersion }}-ci${{ github.run_number }}" | Out-File -FilePath $env:GITHUB_ENV -Encoding utf8 -Append
+        }
+    - name: Build
+      run: dotnet build -p:Version=${{ env.buildVersion }} -p:ContinuousIntegrationBuild=True --no-restore --configuration Release
+    - name: Test
+      run: dotnet test --no-build --configuration Release --verbosity normal
+    - name: Pack
+      if: startsWith(github.ref, 'refs/tags/v')
+      run: dotnet pack -p:PackageVersion=${{ env.nugetVersion }} --configuration Release -o ${{env.DOTNET_ROOT}}/IntelliTect.UtilitiesPack --no-build 
+    - name: Upload Artifacts
+      if: startsWith(github.ref, 'refs/tags/v')
+      uses: actions/upload-artifact@v2
+      with:
+        name: NuGet
+        path: ${{env.DOTNET_ROOT}}/IntelliTect.UtilitiesPack
+      
+  deploy:
+    if: startsWith(github.ref, 'refs/tags/v')
+    runs-on: ubuntu-latest
+    needs: build-and-test
+    environment:
+      name: 'Production'
+    steps:
+      - name: Download artifact from build job
+        uses: actions/download-artifact@v2
+        with:
+          name: NuGet
+      - name: Push NuGet
+        run: |
+          $tagVersion = "${{ github.ref }}".substring(11)
+          echo "tagVersion=$tagVersion" | Out-File -FilePath $env:GITHUB_ENV -Encoding utf8 -Append
+          dotnet nuget push IntelliTect.Utilities.$tagVersion.nupkg --source https://api.nuget.org/v3/index.json -k ${{ secrets.NUGET_API_KEY }} --skip-duplicate
```

Comment:
```suggestion
          dotnet nuget push IntelliTect.Utilities.$tagVersion.nupkg -k ${{ secrets.NUGET_API_KEY }} --skip-duplicate
```

Reply:
I thought this is a necessary parameter? 
https://docs.microsoft.com/en-us/dotnet/core/tools/dotnet-nuget-push

"Starting with NuGet 3.4.2, this is a mandatory parameter unless the NuGet config file specifies a DefaultPushSource value. For more information, see [Configuring NuGet behavior](https://docs.microsoft.com/en-us/nuget/consume-packages/configuring-nuget-behavior)"

Reply:
Ah, I see on mine I have NuGet.config configuring mine. Leave as is